src/os.c

```diff
@@ -237,13 +242,37 @@ void _mi_os_init() {
 
 #else  // generic unix
 
+static void os_detect_transparent_huge_pages(void) {
+#if defined(MADV_HUGEPAGE)
+  int fd = -1;
+  size_t i;
+  static const char *paths[] = {
+      "/sys/kernel/mm/transparent_hugepage/enabled",
+      "/sys/kernel/mm/redhat_transparent_hugepage/enabled",
+  };
+
+  for (i = 0; i < sizeof(paths) / sizeof(paths[0]); i ++) {
+      fd = open(paths[i], O_RDONLY);
+      if (fd > 0) break;
```

Comment:
`open` can return `0`, that's a valid value, not an error.
So I would change that check to `fd >= 0`

Reply:
in general yes even tough stdin might not be closed yet at this early stage but nevertheless I made it less ambiguous.